test/test_parser.cpp

```diff
@@ -2209,13 +2209,14 @@ TEST(Parser_list_of_primitive_ints)
     verify_query(test_context, t2, "list.integers.@avg == 1", 2);
     verify_query(test_context, t2, "list.integers.@sum == 1", 2);
     verify_query(test_context, t2, "list.integers.@min == 1", 2);
+#if 0
     // aggregate operators across multiple lists
     // we haven't supported aggregates across multiple lists previously
     // but the implementation works and applies the aggregate to the primitive list
     // this may be surprising, but it is one way to interpret the expression
     verify_query(test_context, t2, "ALL list.integers == 1", 2);  // row 0 matches {1}. row 1 matches (any of) {} {1}
     verify_query(test_context, t2, "NONE list.integers == 1", 1); // row 1 matches (any of) {}, {0}, {2}, {3} ...
```

Comment:
The old behaviour is to combine all the lists along the chain into one big list, so the explanations for these results are wrong. I think the new behaviour is correct and much easier to reason about. I am in favour of changing this, I just think we should be very clear about it in the changelog in case someone was relying on it.

Reply:
In fact, the right thing to do may be to increase the major version number again. Most SDKs haven't even released v14 yet so this breaking change would be lumped in to all the rest.

Reply:
It is really weird that "ALL {} == 1" matches.

I am not sure we should bump the major. I would call this change a fix. The prior behavior was never really specified, was it? Also the fact that we only had to change  2 checks that had the comment that the result might be surpricing, suggest that this was just "works as implemented".

Reply:
I am quite in doubt how to describe what is breaking here. I think that in the case where you had ANY specified (implicitly or explicitly) and you compare with a single value  or ANY {x,y,z}, then it gives the same results as before. And I think all other cases were broken.

Reply:
That's fair. We can label it a fix, and say we never supported those other many-to-many lists comparisons. I agree that the old behaviour for those cases was strange/broken.